### DISCUSSION TITLE
SharedReader constructor

### DISCUSSION THREAD

**Discussion by jiledaa** [pointfreeco/swift-composable-architecture#3124] [Q&A] [answered]
Hi all. In case I want to use just the synchronisation part of some state but dont want to write to it on certain level of hierarchy, I can use SharedReader, right? But when I have @SharedReader var name: String in state, I cannot just use SharedReader(name) or SharedReader("David") in preview or feature state init  like I am able to do with Shared. I supposed both SharedReader and Shared are the same except SharedReader is not writable. Am I doing something wrong?

OK
```swift
@Shared var name: String
init(name: String) {
    _name = Shared(name)
}
```

nOK (Cannot convert value of type 'String' to expected argument type 'Shared<String>')
```swift
@SharedReader var name: String
init(name: String) {
    _name = SharedReader(name)
}
```

**Comment by mbrandonw** [comment]
Hi @jiledaa, we do not expose that initializer on `SharedReader` because it doesn't seem to be very useful. A `SharedReader(name)` is just a constant that can never change, and so it might as well be a `let name: String`.

Is there a specific use case in mind you have for this?

**Reply by jiledaa** [reply]
Hi, so how is it done when parent has shared state and whenever I change it there, the child updates? But the child never changes the state in reducer, just listens for changes?

**Reply by mbrandonw** [reply, marked as answer]
In that case you need the parent to pass the child the `Shared` state, and then the child can turn it into a `SharedReader`:

```swift
@SharedReader var name: String
init(name: Shared<String>) {
    _name = SharedReader(name)
}
```

**Reply by jiledaa** [reply]
I see, thanks.
